src/lib/InstantSearch.ts

```diff
@@ -616,8 +616,14 @@ See ${createDocumentationLink({
     }
   }
 
+  /**
+   * Set the UI state and trigger a search.
+   * @param uiState The next UI state or a function computing it from the current state
+   * @param fromStateChange private parameter used to know if the method is called from a state change
+   */
   public setUiState(
-    uiState: TUiState | ((previousUiState: TUiState) => TUiState)
+    uiState: TUiState | ((previousUiState: TUiState) => TUiState),
+    callOnStateChange: boolean = true
```

Comment:
I hoped there would not be a need for this parameter, but the only alternative I saw was adding a private setUiState function too, and that seemed overkill (knowing that's callable by users too